### PR TITLE
Fix tests to avoid resetting "compute.default_index_type" and related test failures.

### PR DESCRIPTION
**Pull request by ueshin**
`DefaultIndexTest`s reset the config `compute.default_index_type`, so the following tests are not working under the default index type given by the test environment.

Then found out some tests failed and fixed them (https://github.com/databricks/koalas/runs/1093753632):

- `GroupByTest.test_apply_with_side_effect`
- `NamespaceTest.test_concat_index_axis`

### CODE REVIEW

**Comment by ueshin** [comment]
FYI: testing against `v1.2.0` at https://github.com/ueshin/databricks-koalas/pull/3.

File: databricks/conftest.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ def add_db(doctest_namespace):
     doctest_namespace["db"] = db_name
 
 
-@pytest.fixture(autouse=os.getenv("KOALAS_USAGE_LOGGER", None) is not None)
+@pytest.fixture(autouse=os.getenv("KOALAS_USAGE_LOGGER", "") != "")
```
Comment (itholic) [inline databricks/conftest.py:99]: Is this because "KOALAS_USAGE_LOGGER" can be `None` ??

Comment (ueshin) [reply databricks/conftest.py:99]: It's just changing the way to check the environment variable to see the same between not-set and empty.

Comment (itholic) [reply databricks/conftest.py:99]: Oh I got it, thanks! :)

**Review by itholic — APPROVED** [review]
LGTM

File: databricks/conftest.py (hunk 1 above)
Comment (HyukjinKwon) [inline databricks/conftest.py:99]: Hmm .. why do we need this though? I thought we can just `unset KOALAS_USAGE_LOGGER`.

**Review by HyukjinKwon — APPROVED** [review]
LGTM otherwise

File: databricks/conftest.py (hunk 1 above)
Comment (ueshin) [reply databricks/conftest.py:99]: The variable is always set in CI with conda.
https://github.com/databricks/koalas/blob/bc51b4b06c8db7eca60a3d8f8b31aa4db98642b1/.github/workflows/master.yml#L131
I just wanted to ignore it.

Comment (ueshin) [reply databricks/conftest.py:99]: It's not a big deal. Let me just revert it.

Comment (HyukjinKwon) [reply databricks/conftest.py]: Ah, got it. it should be fine.

Comment (ueshin) [reply databricks/conftest.py]: ah, ok, let's have it.

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**